src/Inventory/Asset/Unmanaged.php

```diff
@@ -0,0 +1,390 @@
+
+    protected $extra_data = [
+        'hardware'        => null,
+        'network_device'  => null,
+    ];
+
+    public function prepare(): array
+    {
+        parent::prepare();
+        $val = $this->data[0];
+
+        $raw_data = $this->raw_data;
+        if (!is_array($raw_data)) {
+            $raw_data = [$raw_data];
+        }
+
+        if (isset($this->extra_data['network_device'])) {
+            $this->prepareForNetworkDevice($val);
+        }
+
+        //remove useless properties
+        // DOMAIN are not managed yet for Unmanaged item
+        if (property_exists($val, 'domains_id')) {
+            unset($val->domains_id);
+        }
+        if (property_exists($val, 'workgroup')) {
+            unset($val->workgroup);
+        }
+
+        $this->data[0] = $val;
+        return $this->data;
+    }
+
+    /**
+     * Prepare network device information
+     *
+     * @param stdClass $val
+     *
+     * @return void
+     */
+    protected function prepareForNetworkDevice($val)
+    {
+        if (isset($this->extra_data['network_device'])) {
+            $device = (object)$this->extra_data['network_device'];
+
+            $dev_mapping = [
+                'mac'       => 'mac',
+                'name'      => 'name',
+                'ips'       => 'ips'
+            ];
+
+            foreach ($dev_mapping as $origin => $dest) {
+                if (property_exists($device, $origin)) {
+                    $device->$dest = $device->$origin;
+                }
+            }
+
+            foreach ($device as $key => $property) {
+                $val->$key = $property;
+            }
+
+            if (property_exists($device, 'ips')) {
+                $portkey = 'management';
+                $port = new \stdClass();
+                if (property_exists($device, 'mac')) {
+                    $port->mac = $device->mac;
+                }
+                $port->name = 'Management';
+                $port->netname = __('internal');
+                $port->instantiation_type = 'NetworkPortAggregate';
+                $port->is_internal = true;
+                $port->logical_number = 0;
+                $port->ipaddress = [];
+
+               //add internal port(s)
+                foreach ($device->ips as $ip) {
+                    if ($ip != '127.0.0.1' && $ip != '::1' && !in_array($ip, $port->ipaddress)) {
+                        $port->ipaddress[] = $ip;
+                    }
+                }
+
+                $this->management_ports[$portkey] = $port;
+            }
+        }
+    }
+
+    /**
+     * After rule engine passed, update task (log) and create item if required
+     *
+     * @param integer $items_id id of the item (0 if new)
+     * @param string  $itemtype Item type
+     * @param integer $rules_id Matched rule id, if any
+     * @param integer $ports_id Matched port id, if any
+     */
+    public function rulepassed($items_id, $itemtype, $rules_id, $ports_id = 0)
```

Comment:
It changed in #13104.
```suggestion
     * @param array $ports_id Matched port ids, if any
     */
    public function rulepassed($items_id, $itemtype, $rules_id, $ports_id = [])
```